eng/pipelines/coreclr/templates/crossgen-comparison-job.yml

```diff
@@ -102,18 +106,29 @@ jobs:
     - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
       - script: |
           chmod +x $(productDirectory)/$(targetFlavor)/$(hostArchType)/crossgen
-          mkdir -p $(artifactsDirectory)/log/$(crossFlavor)
+          mkdir -p $(workItemDirectory)/log/$(crossFlavor)
         displayName: Create directories and ensure crossgen is executable
     - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
       - script: |
-          mkdir $(artifactsDirectory)\log\$(crossFlavor)
+          mkdir $(workItemDirectory)\log\$(crossFlavor)
         displayName: Create directories
 
 
     # Populate Core_Root
     - script: $(coreClrRepoRootDir)build-test$(scriptExt) $(buildConfig) $(archType) $(crossArg) generatelayoutonly
       displayName: Populate Core_Root
 
+    # Populate the workitem directory
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: |
```

Comment:
Why do we need to clone Core_Root? Can we just upload it as is?

Reply:
We need to do copying of some sort, either copy the log folder into core root or both folders to a different location.

Reply:
You can avoid copying at all by appending the log folder to `CorrelationPayloadDirectory` for send-to-helix.yml. You can access the log folder folder on the target machine by using `%HELIX_CORRELATION_PAYLOAD%`

Reply:
Correlation payload is read-only and the job puts the results into the log folder.

Reply:
I cant see a way around the copying, based on @jkoritzinsky this would just need to be copied on the device. Which is in general slower than the copy on the vm.

Reply:
The job can put its own results (for native architecture scenario) inside the work item directory. I am talking about results for cross-architecture scenario that are only read during `compare` step and can be placed inside CorrelationPayloadDirectory .

Anyway, I will let @jashook to decide if he want to do this optimization or not. I am okay with the current state of the changes.